### DISCUSSION TITLE
A question regarding on how to "debug" the board

### DISCUSSION THREAD

**Discussion by DK100** [bitfieldlabs/afterglow#13] [Q&A] [answered]
Hi! I tried to use a self soldered board in a Judge Dredd and it doesn't work. As it is my first time I soldered a SMD board, it is likewise that I did something wrong here. Now I'm wondering on how can I find and solve the issue?
I connected the board using the 18V power plug, R67 is not connected. The green and yellow LEDs on the Arduino are constantly lit. Also all playfield LEDs are lit.
I tried the test mode (switch 1 activated), here the yellow Arduino LED blinks but the playfield LEDs are still constantly lid.
I managed to change the "brown out detection" as mentioned in the wiki.
I would be very thankfull if someone could give me a hints/suggestions on how to figure out, what's wrong here.

![20240106_161308](https://github.com/bitfieldlabs/afterglow/assets/22505495/b272f073-ba4f-4f72-9790-368bca477149)

![20240106_161914](https://github.com/bitfieldlabs/afterglow/assets/22505495/d39e0fd3-1720-4daa-9f68-9f745ab4676b)

**Comment by bitfieldlabs** [comment]
I cannot spot anything obviously wrong on these pictures. The lit/blnking Arduino LED suggests that the firmware is happily working. So the issue is likely with on the driving side, i.e. the 74595 ICs, The ULN and the MOSFETs. Do you have a scope at hand? Then you could have a look at some of the signals in the output line.

**Reply by DK100** [reply]
Thank you very much for your fast reply!
Sadly I don't have a scope. My first thought is to desolder the U5, U6 and U7 and check if the LEDs still light up when switching on the test mode. If so then the MOSFETs must be damaged, correct?
I guess that the U5, U6 and U7 can only be checked all together as rows and columns must be provided together in order to light up an LED?

**Comment by bitfieldlabs** [comment]
I think it's very unlikely that all MOSFETs are defect. I would rather suspect the issue to be with one of the 595s or the ULN. I could only offer to debug it here if you're willing to send it to me.

**Reply by DK100** [reply, marked as answer]
Thank you for the offering to test my board. I tried to resolder the mentioned chips today and as I suspected it looked like I used to much solder in the first run which might have shorted some of the pins. However I made it even worse (I think it took to long and something overheated) and now the Arduino doesn't even start.
I'will give it a second try with a new board.

Anyway, thank you very much for your support and helping me out with your suggestions! Hopefully I have more luck next time ;)
